tools/calyx-writer/lib.rs

```diff
@@ -166,33 +173,31 @@ impl Attribute {
     pub fn bool<S: ToString>(name: S) -> Self {
         Self {
             name: name.to_string(),
-            value: None,
+            value: 1,
         }
     }
 
-    /// Constructs a `calyx_frontend::NumAttr`, such as `@go`, named `name`.    
+    /// Constructs a `calyx_frontend::NumAttr`, such as `@go`, named `name`.
     pub fn num<S: ToString>(name: S, value: u64) -> Self {
         Self {
             name: name.to_string(),
-            value: Some(value),
+            value,
         }
     }
 }
 
 impl CalyxWriter for Attribute {
     fn write(&self, f: &mut IndentFormatter<'_, '_>) -> fmt::Result {
-        write!(f, "@{}", self.name)?;
-        if let Some(value) = self.value {
-            write!(f, "({})", value)?;
-        }
-        Ok(())
+        write!(f, "@{}({})", self.name, self.value)
```

Comment:
It should parse fine but worth noting that `@ATTR(1)`  is the same as `@ATTR`. Take a look at: https://github.com/calyxir/calyx/blob/6c2d82cd5224f32321dd0bd9b7a867e7f0e9c804/calyx-ir/src/printer.rs#L17